### DISCUSSION TITLE
Automatic Sign-in to execute ProvisionComposer.js /Deploy Resources

### DISCUSSION THREAD

**Discussion by UmaPrelude** [microsoft/BotFramework-Composer#5312] [Q&A] [answered]
Hi,

I am using provisioncomposer.js to deploy the resources, but with that we will have to manually sign-in by pasting the url and code. Is there an option to use service principal / managed identity /connect command to automatically login from some configuration.
(I am new to those concepts, please correct me if i am wrong somewhere)

I have configured provisioncomposer.js in Jenkins. I want to make it automated for creating the resources and update the appsettings.json with the generated keys.

If i get some support, it will be great.

- UmPrel

**Comment by tonyanziano** [comment]
@benbrown can you please comment on this?

**Comment by benbrown** [comment, marked as answer]
This provision script using ARM templates and the az cli to achieve the outcome.  It is possible to use your own mechanism and auth to execute the ARM templates, but we do not plan to add this functionality natively to the provision script.

**Comment by UmaPrelude** [comment]
Hi Ben,

I noticed the below thread where it is mentioned that composer team will provide some support for CI/CD related documentation.
https://github.com/microsoft/BotFramework-Composer/issues/3339
If that is the case, there should be some option for the automated sign-in i felt.
Any comments on this.

**Comment by benbrown** [comment]
@UmaPrelude in the ci/cd environment, credentials are provided via different mechanisms other than interactive login. Future integrations in that area will take advantage of the available auth mechanisms - keyvault, service principals, etc.

**Reply by UmaPrelude** [reply]
> @UmaPrelude in the ci/cd environment, credentials are provided via different mechanisms other than interactive login. Future integrations in that area will take advantage of the available auth mechanisms - keyvault, service principals, etc.

Thanks for the response ben. Any expected timeline for those releases?
